modules/fedimint-gw-client/src/pay.rs

```diff
@@ -0,0 +1,17 @@
+/// State machine that pays a lightning invoice on behalf of a federation user.
+///
+/// ```mermaid
+/// graph LR
+/// classDef virtual fill:#fff,stroke-dasharray: 5 5
+///
+///    PayInvoice -- validate outgoing contract --> Canceled
+///    PayInvoice -- await preimage offer accepted --> BuyPreimage
+///    PayInvoice -- await preimage offer failure --> Canceled
+///    BuyPreimage -- await federation decryption --> Preimage
+///    BuyPreimage -- external payment --> Preimage
+///    BuyPreimage -- await buy preimage failure --> Refund
+///    Preimage -- await claim outgoing contract --> Outpoint
+///    Preimage -- await claim outgoing contract failure --> Refund
```

Comment:
How do we get a refund in this case? Seems more like a hard error condition.

Reply:
Agreed, this should be a failure state